src/components/dashboards/Admin-dashboard.tsx

```diff
@@ -1,20 +1,100 @@
 "use client";
 import { useSession } from "next-auth/react";
 import { useRouter } from "next/navigation";
+import { useEffect, useState } from "react";
+
+interface TimeSlot {
+  id: string;
+  startTime: string;
+  endTime: string;
+  description: string;
+}
+
+interface Ticket {
+  id: string;
+  status: string;
+  requestedTimeSlot: TimeSlot;
+  user: User;
+}
+
+interface User {
+  id: string;
+  isAdmin: boolean;
+  email: string;
+  phoneNumber?: string; // Optional if it can be null
+  name: string;
+}
 
 export default function AdminDashboard() {
   const router = useRouter();
   const { data: session, status } = useSession();
+  const [openTickets, setOpenTickets] = useState<Ticket[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  // Redirect unauthenticated users
+  useEffect(() => {
+    if (status === "unauthenticated") {
+      router.replace("/signup");
+    }
+  }, [status, router]);
+
+  // Fetch open tickets when the user is an admin
+  useEffect(() => {
+    if (session?.user?.isAdmin) {
+      fetch("/api/tickets?status=PENDING")
+        .then((response) => response.json())
+        .then((data) => {
+          setOpenTickets(data);
+          console.log(data);
+          setLoading(false);
+        })
+        .catch((error) => {
+          console.error("Error fetching tickets:", error);
+          setError("Failed to load tickets. Please try again later.");
+          setLoading(false);
+        });
+    }
+  }, [session?.user?.isAdmin]);
```

Comment:
I think we need to have a tech discussion w/ the dev team on whether or not we want to have fetching logic at the component level instead of, say, in a service?  I prefer it in a separate file with similar related logic, as it's in one place.
Thoughts?